adobe/react-spectrum
# fix(#2659): useSliderThumb: fire onChangeEnd for PageUp/PageDown/Home/End

Closes #2659 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue 2659](https://github.com/adobe/react-spectrum/issues/2659).

## 📝 Test Instructions:

1. Open Slider Hooks Single Story: https://reactspectrum.blob.core.windows.net/reactspectrum/92cb6eea1c50f026636dd7e3ae9f8cb7c57ab6d5/storybook/index.html?path=/story/slider-hooks--single
2. Navigate to focus the Slider example.
3. Press ArrowRight to increment the value by 1. Notice in Actions panel, both onChange and onChangeEnd fire.
4. Press ArrowLeft to decrement the value by 1. Notice in Actions panel, both onChange and onChangeEnd fire.
5. With slider at the minimum value, 0, press ArrowLeft to decrement the value by 1. Notice in Actions panel, only onChangeEnd fires, because the value has not changed.
6. Press End key (fn+ArrowRight on a macOS laptop), to jump to the maximum value. Notice in Actions panel, both onChange fires and onChangeEnd fire.
7. With slider at the maximum value, 100, press ArrowRight to increment the value by 1. Notice in Actions panel, only onChangeEnd fires, because the value has not changed.
8. Press Home key (fn+ArrowLeft on a macOS laptop), to jump to the minimum value. Notice in Actions panel, both onChange and onChangeEnd fire.
9. Press PageUp key (fn+ArrowUp on a macOS laptop), to increment by 10. Notice in Actions panel, both onChange and onChangeEnd fire.
10. Press PageUp key (fn+ArrowDown on a macOS laptop), to decrement by 10. Notice in Actions panel, both onChange and onChangeEnd fire.
11. Increment or decrement by step using the arrow keys. Notice in Actions panel, both onChange and onChangeEnd fire for each arrow key press.

## 🧢 Your Project:

Adobe/Accessibility
